### PR TITLE
Add --output-orphaned option to split-paired-reads.py

### PR DESCRIPTION
**Pull request by camillescott**
Currently, so far as I can tell, there's no way to both split paired reads into .1 and .2 files while also outputting orphans to their own file without doing messy pipe stuff. This adds that feature to split-paired-reads.py with an extra command line option.

### CODE REVIEW

**Comment by ctb** [comment]
This should fix #847, right?

**Comment by ctb** [comment]
@camillescott may I snarf this?

**Comment by ctb** [comment]
ping @camillescott

**Comment by ctb** [comment]
Taking it and making it my own.

**Comment by mr-c** [comment]
LGTM

**Comment by ctb** [comment]
- [ ] streaming_io tests
- [ ] maybe add in gzip stuff from #747

**Comment by ctb** [comment]
- [ ] test -0 without filename
- [ ] test -0 without filename and with output-directory
- [ ] test -0 with filename
- [ ] test -p and evaluat eerror message
- [ ] compression stuff (bzip2/gzip)

**Comment by ctb** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Is the Copyright year up to date?

Ready for review @mr-c @bocajnotnef @luizirber

**Comment by ctb** [comment]
Fixes #847.

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,14 @@
+2015-08-04  Titus Brown  <titus@idyll.org> and
+Camille Scott  <camille.scott.w@gmail.com>
```
Comment (mr-c) [inline ChangeLog]: Can you indent this to match your name?

File: scripts/split-paired-reads.py [hunk 2]
```diff
@@ -37,16 +38,16 @@ def get_parser():
...
 
     Alternatively, you can specify the filenames directly with
     :option:`-1`/:option:`--output-first` and
     :option:`-2`/:option:`--output-second`, which will override the
-    :option:`-o`/:option:`--output-dir` setting on a file-specific basis.
+    :option:`-d`/:option:`--output-dir` setting on a file-specific basis.
 
-    :option:`-p`/:option:`--force-paired` will require the input file to
-    be properly interleaved; by default, this is not required.
+    :option:`-0`/:option:'`--output-orphans` will allow broken-paired format,
```
Comment (mr-c) [inline scripts/split-paired-reads.py]: extra extraneous single quote after the 2nd `:option:`

File: scripts/split-paired-reads.py [hunk 3]
```diff
@@ -128,47 +131,48 @@ def main():
         # Use default filename created above
         fp_out2 = get_file_writer(open(out2, 'wb'), args.gzip, args.bzip)
 
+    # put orphaned reads here, if -0!
+    allow_orphans = False
```
Comment (mr-c) [inline scripts/split-paired-reads.py]: Why not use `args.output_orphaned` everywhere you check `allow_orphans`?

**Comment by mr-c** [comment]
Otherwise, this LGTM

File: scripts/split-paired-reads.py (hunk 3 above)
Comment (ctb) [inline scripts/split-paired-reads.py]: On Tue, Aug 04, 2015 at 09:52:19AM -0700, Michael R. Crusoe wrote:
> > @@ -128,47 +131,48 @@ def main():
> >          # Use default filename created above
> >          fp_out2 = get_file_writer(open(out2, 'wb'), args.gzip, args.bzip)
> >  
> > +    # put orphaned reads here, if -0!
> > +    allow_orphans = False
> 
> Why not use `args.output_orphaned` everywhere you check `allow_orphans`?

It's partly a leftover of some now-removed logic that permitted -0 without
a filename, but what made me keep it in was this code:

`require_paired=not allow_orphans`

which I think would be uglier and less clear if it was
`not args.output_orphaned`.  What do you think?

File: scripts/split-paired-reads.py (hunk 2 above)
Comment (ctb) [inline scripts/split-paired-reads.py]: On Tue, Aug 04, 2015 at 09:50:34AM -0700, Michael R. Crusoe wrote:
> >  
> > -    :option:`-p`/:option:`--force-paired` will require the input file to
> > -    be properly interleaved; by default, this is not required.
> > +    :option:`-0`/:option:'`--output-orphans` will allow broken-paired format,
> 
> extra extraneous single quote after the 2nd `:option:`

how do you find this stuff!? fixed :)

**Comment by ctb** [comment]
- [ ] standardize on --output instead of --out (see #1188)

File: scripts/split-paired-reads.py (hunk 3 above)
Comment (mr-c) [inline scripts/split-paired-reads.py]: I'm okay with `not args.output_orphaned`

Comment (ctb) [inline scripts/split-paired-reads.py]: On Tue, Aug 04, 2015 at 03:54:24PM -0700, Michael R. Crusoe wrote:
> > @@ -128,47 +131,48 @@ def main():
> >          # Use default filename created above
> >          fp_out2 = get_file_writer(open(out2, 'wb'), args.gzip, args.bzip)
> >  
> > +    # put orphaned reads here, if -0!
> > +    allow_orphans = False
> 
> I'm okay with `not args.output_orphaned` 

That doesn't address "here's my reasoning, what do you think" but I'll give
it a whirl ;)

**Comment by ctb** [comment]
Fixes #1188.

Ready for review @mr-c.

**Comment by mr-c** [comment]
Great, thanks!

**Pull request merged**